wallet-new/server/Cardano/Wallet/API/V1/LegacyHandlers/Transactions.hs

```diff
@@ -51,36 +52,43 @@ newTransaction submitTx Payment {..} = do
 
 allTransactions
     :: forall ctx m. (V0.MonadWalletHistory ctx m)
-    => WalletId
+    => Maybe WalletId
     -> Maybe AccountIndex
     -> Maybe (V1 Core.Address)
     -> RequestParams
     -> FilterOperations Transaction
     -> SortOperations Transaction
     -> m (WalletResponse [Transaction])
-allTransactions walletId mAccIdx mAddr requestParams fops sops = do
-    cIdWallet <- migrate walletId
-    ws <- V0.askWalletSnapshot
-
-    -- Create a `[V0.AccountId]` to get txs from it
-    let accIds = case mAccIdx of
-            Just accIdx -> migrate (walletId, accIdx)
-            -- ^ Migrate `V1.AccountId` into `V0.AccountId` and put it into a list
-            Nothing     -> V0.getWalletAccountIds ws cIdWallet
-            -- ^ Or get all `V0.AccountId`s of a wallet
+allTransactions mwalletId mAccIdx mAddr requestParams fops sops  =
+    case mwalletId of
+        Just walletId -> do
+            cIdWallet <- migrate walletId
+            ws <- V0.askWalletSnapshot
 
-    let v0Addr = case mAddr of
-            Nothing        -> Nothing
-            Just (V1 addr) -> Just $ V0.encodeCType addr
+            -- Create a `[V0.AccountId]` to get txs from it
+            let accIds = case mAccIdx of
+                    Just accIdx -> migrate (walletId, accIdx)
+                    -- ^ Migrate `V1.AccountId` into `V0.AccountId` and put it into a list
+                    Nothing     -> V0.getWalletAccountIds ws cIdWallet
+                    -- ^ Or get all `V0.AccountId`s of a wallet
 
-    -- get all `[Transaction]`'s
-    let transactions = do
-            (V0.WalletHistory wh, _) <- V0.getHistory cIdWallet (const accIds) v0Addr
-            migrate wh
+            let v0Addr = case mAddr of
+                    Nothing        -> Nothing
+                    Just (V1 addr) -> Just $ V0.encodeCType addr
 
-    -- Paginate result
-    respondWith requestParams fops sops (IxSet.fromList <$> transactions)
+            -- get all `[Transaction]`'s
+            let transactions = do
+                    (V0.WalletHistory wh, _) <- V0.getHistory cIdWallet (const accIds) v0Addr
+                    migrate wh
 
+            -- Paginate result
+            respondWith requestParams fops sops (IxSet.fromList <$> transactions)
+        _ ->
+            -- TODO: we need to implement the case for Nothing
+            -- TODO: should we use the 'FilterBy' machinery instead? that
+            --       let us express RANGE, GT, etc. in addition to EQ. does
+            --       that make sense for this dataset?
+            error "Implement me!"
```

Comment:
Looks like the old code requests *either* a `wallet_id` *or* an `AccountId`, which is a wallet ID paired with an account index:

```haskell
data AccountId = AccountId
    { -- | Address of wallet this account belongs to
      aiWId   :: CId Wal
    , -- | Derivation index of this account key
      aiIndex :: Word32
    } deriving (Eq, Ord, Show, Generic, Typeable)
```

So a `WalletId` is in fact necessary -- and we optionally can take an `AccountIndex`. Which is what the original API supported :\

Reply:
However, we're aiming to support eg `BlockExplorer` use case, which gets 5 most recent transactions that the wallet knows about, regardless of who is doing it.

So we can keep the query param, do a `throwError err400` if they don't pass it, and later un-break that part of the API when we can support it.